app/session.go

```diff
@@ -242,29 +241,24 @@ func (a *App) UpdateLastActivityAtIfNeeded(session model.Session) {
 }
 
 func (a *App) CreateUserAccessToken(token *model.UserAccessToken) (*model.UserAccessToken, *model.AppError) {
-	if !*a.Config().ServiceSettings.EnableUserAccessTokens {
+
+	user, err := a.Srv.Store.User().Get(token.UserId)
+
+	if !*a.Config().ServiceSettings.EnableUserAccessTokens && !user.IsBot {
```

Comment:
Couldn't `user` be a nil here, since there is no immediate error check?

Reply:
Should the err value in this call cover the nil user case?  Looking at the previous implementation, it looks like user == nil was not checked.   I would like to locate the Get Function call, in the code base, but haven't found it yet.  Where is this method defined?

The err check is later in the function on rows 259, 260, but I think it should be moved under the call to create the user (line 245).

Reply:
@levb Thoughts on the above note from @jfrerich?

Reply:
@jfrerich What are you thoughts on this?

Reply:
@hanzei I agree with Lev and will get the change added and resubmit the PR.